fix(tui): add missing Ctrl+n/Ctrl+p support to ListSelectionView

pppp606 opened the pull request:
## Summary

  Extend Ctrl+n/Ctrl+p navigation support to selection popups (model picker, approval mode, etc.)

  This is a follow-up to #7530, which added Ctrl+n/Ctrl+p navigation to the textarea.
  The same keybindings were missing from `ListSelectionView`, causing inconsistent behavior
  when navigating selection popups.

  ## Related

  - #7530 - feat(tui): map Ctrl-P/N to arrow navigation in textarea

  ## Changes

  - Added Ctrl+n as alternative to Down arrow in selection popups
  - Added Ctrl+p as alternative to Up arrow in selection popups
  - Added unit tests for the new keybindings

  ## Test Plan

  - [x] `cargo test -p codex-tui list_selection_view` - all tests pass
  - [x] Manual testing: verified Ctrl+n/p navigation works in model selection popup

pppp606 committed:
feat(tui): add Ctrl+n/Ctrl+p navigation to selection popups

Extends keyboard navigation support to approval mode, model selection,
and rate limit switch popups, matching the behavior added in #1994
for slash commands and file selector.

pppp606 commented:
I have read the CLA Document and I hereby sign the CLA

pppp606 committed:
fix(tui): handle C0 control chars for Ctrl+n/Ctrl+p in selection popups

Some terminals send Control key chords as C0 control characters
without reporting the CONTROL modifier. Add fallback handling for
^N (U+000E) and ^P (U+0010) to ensure consistent navigation across
all terminal environments.

Follows the same pattern used in textarea.rs (see #7530).

pppp606 commented on codex-rs/tui/src/bottom_pane/list_selection_view.rs:
```diff
@@ -266,10 +266,20 @@ impl BottomPaneView for ListSelectionView {
+                code: KeyCode::Char('p'),
+                modifiers: KeyModifiers::CONTROL,
+                ..
             } => self.move_up(),
```
fix

[a154d28](https://github.com/openai/codex/pull/7629/commits/a154d281040afffe99b59ab63812633c52afff9d)

etraut-openai reviewed (changes requested):
Thanks for the PR. Looks good for the most part. 

While we like regression tests in the general case, I think these are unnecessary in this specific case.

pppp606 committed:
refactor: remove Ctrl+n/Ctrl+p tests per review feedback

Tests for this functionality are considered overkill.
Removed to keep CI times in check.

pppp606 committed:
Merge branch 'main' into feat/ctrl-np-navigation

etraut-openai committed:
Merge branch 'main' into feat/ctrl-np-navigation
